### DISCUSSION TITLE
Confusing withAuth behaviour

### DISCUSSION THREAD

**Discussion by crowncode-development** [keystonejs/keystone#8798] [Questions] [answered]
We are using Keystone with Next.JS and have a requirement to store session data for unauthenticated users. Additionally, we need to verify this data on a different URL.

And follow the examples we use something liks:

`/api/url1.js`
```
export default async function handler(req, res) {
    ...
    const someData = { ... }
    const context = await keystoneContext.withRequest(req, res)
    await context.sessionStrategy.start({
      context,
      data: { someData },
    })
    ...
}
```

`/api/url2.js`
```
export default async function handler(req, res) {
    ...
    const context = await keystoneContext.withRequest(req, res)
    if (context.session.someData) {
        // do some logic
    }
    ...
}
```

And this example is not working! Because of [this code](https://github.com/keystonejs/keystone/blob/19bb460/packages/auth/src/index.ts#L169-L190):

<img width="808" alt="image" src="https://github.com/keystonejs/keystone/assets/137823147/67cd26f4-703e-4b7b-97ac-1ee2149bbeba">

As you can see here, if you try to save any session data without providing an itemId and listKey, you will always receive an empty context.session.

And it's little confusing me. I would expect to receive the data that I have saved. Can you change it?

**Comment by borisno2** [comment]
Hi @crowncode-development 

I agree this is not ver intuitive and we are looking at ways we can back the session API better, in the mean time our 
[custom session invalidation example](https://github.com/keystonejs/keystone/blob/main/examples/custom-session-invalidation/keystone.ts) shows a way you can add and retrieve data from the initial session object before it goes through with.you might still need some kind of guest list item in you user table though so the auth package respects the session.

**Comment by dcousens** [comment, marked as answer]
In short, the `@keystone-6/auth`, as it stands today, is not designed for other usecases.  It is highly opinionated.
We intend to decrease the surface area of the package in time, while adding support to `@keystone-6/core` for easier ways to compose different authentication strategies like what you are trying to do.
